src/MCPClient/lib/settings/common.py

```diff
@@ -16,32 +16,186 @@
+
+    def get(self, attr, default=None):
+        if attr not in self.MAPPING:
+            raise ImproperlyConfigured('Unknown attribute: %s. Make sure the attribute is included in the MAPPING property in `common.py`.' % attr)
+
+        attr_opts = self.MAPPING[attr]
+        if not all(k in attr_opts for k in ('section', 'option', 'type')):
+            raise ImproperlyConfigured('Invalid attribute: %s. Make sure the entry in MAPPING in common.py has all the fields needed (section, option, type).' % attr)
+
+        getter = 'get{}'.format('' if attr_opts['type'] == 'string' else attr_opts['type'])
+        kwargs = {'section': attr_opts['section'], 'option': attr_opts['option']}
+        if default is not None:
+            kwargs['fallback'] = default
+        elif 'default' in attr_opts:
+            kwargs['fallback'] = attr['default']
+
+        try:
+            return getattr(self.config, getter)(**kwargs)
+        except (ConfigParser.NoSectionError, ConfigParser.NoOptionError):
+            raise ImproperlyConfigured('The following configuration attribute must be defined: %s.' % attr)
+
+
+config = Config()
 
-# Get DB settings from main configuration file
-config = ConfigParser.SafeConfigParser({'max_overflow': 40})
-config.read('/etc/archivematica/archivematicaCommon/dbsettings')
 
 DATABASES = {
     'default': {
-        'ENGINE': 'django_mysqlpool.backends.mysqlpool',
-        'NAME': config.get('client', 'database'),          # Or path to database file if using sqlite3.
-        'USER': config.get('client', 'user'),              # Not used with sqlite3.
-        'PASSWORD': config.get('client', 'password'),      # Not used with sqlite3.
-        'HOST': config.get('client', 'host'),              # Set to empty string for localhost. Not used with sqlite3.
-        'PORT': '',                                        # Set to empty string for default. Not used with sqlite3.
+        'ENGINE': config.get('db_engine'),
+        'NAME': config.get('db_name'),
+        'USER': config.get('db_user'),
+        'PASSWORD': config.get('db_password'),
+        'HOST': config.get('db_host'),
+        'PORT': config.get('db_port'),
     }
 }
 
 MYSQLPOOL_BACKEND = 'QueuePool'
 MYSQLPOOL_ARGUMENTS = {
     'use_threadlocal': False,
-    'max_overflow': config.get('client', 'max_overflow'),
+    'max_overflow': config.get('db_pool_max_overflow'),
 }
 
 CONN_MAX_AGE = 14400
 
 # Make this unique, and don't share it with anybody.
-SECRET_KEY = 'e7b-$#-3fgu)j1k01)3tp@^e0=yv1hlcc4k-b6*ap^zezv2$48'
+SECRET_KEY = config.get('secret_key', default='e7b-$#-3fgu)j1k01)3tp@^e0=yv1hlcc4k-b6*ap^zezv2$48')
 
 USE_TZ = True
 TIME_ZONE = 'UTC'
+
+INSTALLED_APPS = (
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'components.accounts',
+    'main',
+    'components.mcp',
+    'components.administration',
+    'fpr',
+)
```

Comment:
The client scripts interact directly with the Dashboard models so we have to make sure that all the apps are installed. Before the client scripts just used to make use of the Dashboard settings which was even worse.